[KF-6782] Adding bundle version check to UATs

deusebio opened the pull request:
The PR adds a check to make sure that the Kubeflow deployment has the correct, consistent versions. 

The PR has been tested in the KF bundle tests on [AWS EKS](https://github.com/canonical/bundle-kubeflow/actions/runs/12946023787/job/36122354217)

misohu commented on driver/test_kubeflow_workloads.py:
```diff
@@ -59,28 +60,62 @@
 
 KFP_PODDEFAULT_NAME = "access-ml-pipeline"
 
+CHARMS = {
```
Cant we get this json from bundle-kubeflow repository https://github.com/canonical/bundle-kubeflow/tree/main/releases/1.9/stable instead ? By having it here we may have problems in the future when we decide to do some changes in the bundle file (we might forget to update this one).

misohu reviewed (commented):
One small comment. Can we also link jira issue in the description.

deusebio commented:
@misohu please re-review. I have addressed your comment and made sure that the UAT still works by triggering another run on [EKS](https://github.com/canonical/bundle-kubeflow/actions/runs/12946023787/job/36446345510)

misohu reviewed (approved):
Nice job thx
